### PR TITLE
Add the ability to configure slack markdown field

### PR DESCRIPTION
**Pull request by asherf**
Closes https://github.com/prometheus/alertmanager/issues/1220.

### CODE REVIEW

File: config/notifiers_test.go [hunk 1]
```diff
@@ -330,7 +330,72 @@ token: ''
...
+			t.Errorf("\nexpected:\n%v\ngot:\n%v", rt.expected.MrkdwnIn, cfg.MrkdwnIn)
+		}
+		for i := range cfg.MrkdwnIn {
+			if rt.expected.MrkdwnIn[i] != cfg.MrkdwnIn[i] {
+				t.Errorf("\nexpected:\n%v\ngot:\n%v\nat index %v", rt.expected.MrkdwnIn[i], cfg.MrkdwnIn[i], i)
+			}
+		}
 
+	}
+
```
Comment (simonpasquier) [inline config/notifiers_test.go]: could you remove this empty line and add one between L398 and L399? Otherwise LGTM.

Comment (asherf) [reply config/notifiers_test.go]: done.

**Comment by asherf** [comment]
I am happy to add more test cases or more tests for this if this is need....
any other feedback is welcome as I am new to Go and this code base.

**Comment by simonpasquier** [comment]
Thanks!

**Comment by DeathBorn** [comment]
But everybody forgot to add this to documentation :(

**Comment by asherf** [comment]
@DeathBorn I at least tried... ;-) ... https://github.com/prometheus/docs/pull/1387

**Pull request merged**